flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobServer.java

```diff
@@ -354,10 +354,12 @@ public void close() throws IOException {
         if (shutdownRequested.compareAndSet(false, true)) {
             Exception exception = null;
 
-            try {
-                this.serverSocket.close();
-            } catch (IOException ioe) {
-                exception = ioe;
+            if (serverSocket != null) { // can be null if shut down before constructor completion
```

Comment:
hm, wouldn't it be more consistent to move the shutdown hook initialization (along with the cleanup scheduling) at the end of the `BlobServer` initialization?

It feels wrong to register the shutdown hook for an instance that's not fully created, yet. 🤔

Reply:
hm, I guess, you're right. The socket creation should live in the `BlobServer#start` method. ...from a conceptual standpoint.

Shall we make the `serverSocket` field `@Nullable` as part of this change to underline this?

Reply:
Good point, added annotation.